OJS.Workers.ExecutionStrategies/BaseExecutionStrategy.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System;
+    using System.Threading.Tasks;
+
+    using log4net;
+
+    using OJS.Workers.Common;
+    using OJS.Workers.Common.Exceptions;
+    using OJS.Workers.Common.Helpers;
+    using OJS.Workers.ExecutionStrategies.Models;
+
+    public abstract class BaseExecutionStrategy : IExecutionStrategy
+    {
+        private readonly ILog logger;
+
+        public BaseExecutionStrategy() => this.logger = LogManager.GetLogger(typeof(BaseExecutionStrategy));
+
+        protected string WorkingDirectory { get; set; }
+
+        public IExecutionResult<TResult> Execute<TInput, TResult>(IExecutionContext<TInput> executionContext)
+            where TResult : ISingleCodeRunResult, new()
+            => this.InternalExecute(executionContext, new ExecutionResult<TResult>());
+
+        public IExecutionResult<TResult> SafeExecute<TInput, TResult>(IExecutionContext<TInput> executionContext)
+            where TResult : ISingleCodeRunResult, new()
+        {
+            this.WorkingDirectory = DirectoryHelpers.CreateTempDirectoryForExecutionStrategy();
+
+            try
+            {
+                return this.Execute<TInput, TResult>(executionContext);
+            }
+            finally
```

Comment:
Is lack of `catch` on purpose? If so consider adding a brief comment, because it seems a bit odd. As it is now, will it not just exit the app silently, after attempting to delete the directory?

Reply:
Yes it is. The deletion of the directory should not stop the flow. I added comments.